specification/applicationinsights/ApplicationInsights.LiveMetrics/models.tsp

```diff
@@ -0,0 +1,567 @@
+  DocumentStreamIds?: string[];
+
+  @doc("Collection of custom properties.")
+  Properties?: KeyValuePairStringString[];
+}
+
+/**
+ * Request document type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model Request extends DocumentIngress {
+  @doc("Telemetry type for Request.")
+  DocumentType: "Request";
+
+  @doc("Name of the request, e.g., 'GET /values/{id}'.")
+  @maxLength(1024)
+  Name?: string;
+
+  @doc("Request URL with all query string parameters.")
+  @maxLength(2048)
+  Url?: string;
+
+  @doc("Result of a request execution. For http requests, it could be some HTTP status code.")
+  @maxLength(1024)
+  ResponseCode?: string;
+
+  @doc("Request duration in ISO 8601 duration format, i.e., P[n]Y[n]M[n]DT[n]H[n]M[n]S or P[n]W.")
+  Duration?: string;
+}
+
+/**
+ * RemoteDependency document type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model RemoteDependency extends DocumentIngress {
+  @doc("Telemetry type for RemoteDependency.")
+  DocumentType: "RemoteDependency";
+
+  @doc("Name of the command initiated with this dependency call, e.g., GET /username.")
+  @maxLength(1024)
+  Name?: string;
+
+  @doc("URL of the dependency call to the target, with all query string parameters.")
+  @maxLength(2048)
+  CommandName?: string;
+
+  @doc("Result code of a dependency call. Examples are SQL error code and HTTP status code.")
+  @maxLength(1024)
+  ResultCode?: string;
+
+  @doc("Request duration in ISO 8601 duration format, i.e., P[n]Y[n]M[n]DT[n]H[n]M[n]S or P[n]W.")
+  Duration?: string;
+}
+
+/**
+ * Exception document type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model Exception extends DocumentIngress {
+  @doc("Telemetry type for Exception.")
+  DocumentType: "Exception";
+
+  @doc("Exception type name.")
+  @maxLength(1024)
+  ExceptionType?: string;
+
+  @doc("Exception message.")
+  @maxLength(32768)
+  ExceptionMessage?: string;
+}
+
+/**
+ * Event document type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model Event extends DocumentIngress {
+  @doc("Telemetry type for Event.")
+  DocumentType: "Event";
+
+  @doc("Event name.")
+  @maxLength(512)
+  Name?: string;
+}
+
+/**
+ * Trace document type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model Trace extends DocumentIngress {
+  @doc("Telemetry type for Trace.")
+  DocumentType: "Trace";
+
+  @doc("Trace message.")
+  @maxLength(32768)
+  Message?: string;
+}
+
+/**
+ * Key-value pair of string and string.
+ */
+model KeyValuePairStringString {
```

Comment:
This name is definitely going to cause concern in the SDK archboard review. Wherever you use it, you can also use `Record<string>` which should provide the same structure.